packages/fx-core/src/component/driver/aad/interface/createAadAppOutput.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+export interface CreateAadAppOutput {
+  // Non secrets
+  AAD_APP_CLIENT_ID?: string;
```

Comment:
Are these fields optional by design?

Reply:
Yes, though it was called output, it also serves as input because we're doing create if not exist things which relies on previous output.
You can think what the action does is: if one property in output is empty, it will set the property's value and output to .env. There's one exception that when client id exists, the action will not fill missing values.